### PR TITLE
III-6345 Reuse existing role

### PR DESCRIPTION
**Pull request by LucWollants**
### Added
- Added `role_id` to table `ownership_search`
- Added method `updateRoleId` to set the link between an ownership and a role

### Changed
- When approving an ownership check for an existing role for the organizer, if this exists add the user to this role
- When deleting an ownership delete the user from the linked role

---

Ticket: https://jira.uitdatabank.be/browse/III-6345

### CODE REVIEW

**Review by grubolsch — CHANGES_REQUESTED** [review]

File: src/Ownership/Readmodels/OwnershipPermissionProjector.php [hunk 1]
```diff
@@ -60,32 +58,46 @@ public function handle(DomainMessage $domainMessage): void
 
     protected function applyOwnershipApproved(OwnershipApproved $ownershipApproved): void
     {
+        $this->commandBus->disableAuthorization();
+
         $ownershipItem = $this->ownershipSearchRepository->getById($ownershipApproved->getId());
+        $roleId = $this->getExistingRoleId($ownershipItem->getItemId());
 
-        $this->commandBus->disableAuthorization();
+        if ($roleId !== null) {
```
[the remark this answers is not in the document]
Comment (LucWollants) [reply src/Ownership/Readmodels/OwnershipPermissionProjector.php:66]: I think that is already done?

File: src/Ownership/Readmodels/OwnershipPermissionProjector.php [hunk 2]
```diff
@@ -94,19 +106,45 @@ protected function applyOwnershipDeleted(OwnershipDeleted $ownershipDeleted): vo
     {
         $this->commandBus->disableAuthorization();
 
-        $roles = $this->roleSearchRepository->search(
-            $this->createRoleName($ownershipDeleted->getId())
-        );
+        $ownershipItem = $this->ownershipSearchRepository->getById($ownershipDeleted->getId());
 
-        foreach ($roles->getMember() as $role) {
+        if ($ownershipItem->getRoleId()) {
```
[the remark this answers is not in the document]
Comment (LucWollants) [reply src/Ownership/Readmodels/OwnershipPermissionProjector.php]: Good suggestion, I changed that.

File: src/Ownership/Readmodels/OwnershipPermissionProjector.php [hunk 3]
```diff
@@ -94,19 +106,45 @@ protected function applyOwnershipDeleted(OwnershipDeleted $ownershipDeleted): vo
...
+                0,
+                1
+            )
+        );
+
+        if ($existingOwnerships->count() < 1) {
+            return null;
+        }
+
+        if ($existingOwnerships->getFirst()->getRoleId() === null) {
```
[the remark this answers is not in the document]
Comment (LucWollants) [reply src/Ownership/Readmodels/OwnershipPermissionProjector.php]: Removed his line

File: src/Ownership/Readmodels/OwnershipPermissionProjector.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (grubolsch) [reply src/Ownership/Readmodels/OwnershipPermissionProjector.php:66]: Okey, can we just change the second if into an else then? I will already approve

[the remark this answers is not in the document]
Comment (LucWollants) [reply src/Ownership/Readmodels/OwnershipPermissionProjector.php:66]: Done

**Pull request merged**